datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java

```diff
@@ -97,7 +101,33 @@ public static BigQuery getClientForServiceAccount(
   public static TableResult queryBigQuery(BigQuery bigQueryClient, String query)
       throws InterruptedException {
     QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(query).build();
-    return bigQueryClient.query(queryConfig);
+    BigQueryResult bigQueryResult = new BigQueryResult(bigQueryClient, queryConfig);
+    Awaitility.await().atMost(17, TimeUnit.MINUTES).until(bigQueryResult);
```

Comment:
Is there a rationale behind 17 minutes here? If so, would you mind adding a comment?